### PR TITLE
💥 [RUMF-730] prefer object and type alias over enum in APIs

### PR DESCRIPTION
**Pull request by bcaudan**
## Motivation

make APIs easier to use with TypeScript by avoiding to import our enums

before
```ts
import { datadogRum, Datacenter } from '@datadog/browser-rum'

datadogRum.init({
  ...
  datacenter: Datacenter.US
  ...
})
```

after
```ts
import { datadogRum } from '@datadog/browser-rum'

datadogRum.init({
  ...
  datacenter: 'us'
  ...
})
```

## Changes

replace enum usage in APIs by object and type alias

before
```ts
export enum Datacenter {	
  US = 'us',	  
  EU = 'eu',	  
}
```

after
```ts
export const Datacenter = {
  US: 'us',	  
  EU: 'eu',	  
} as const

export type Datacenter = (typeof Datacenter)[keyof typeof Datacenter]
```

## Testing

ci

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/master/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/logs/src/boot/logs.entry.ts [hunk 1]
```diff
@@ -77,7 +75,7 @@ export function makeLogsGlobal(startLogsImpl: StartLogs) {
     removeLoggerGlobalContext: monitor(globalContextManager.remove),
 
     createLogger: monitor((name: string, conf: LoggerConfiguration = {}) => {
-      customLoggers[name] = new Logger(sendLog, conf.handler, conf.level, {
+      customLoggers[name] = new Logger(sendLog, conf.handler as HandlerType, conf.level as StatusType, {
```
Comment (BenoitZugmeyer) [inline packages/logs/src/boot/logs.entry.ts:78]: What about using constant unions internally too? to avoid those cast

**Comment by BenoitZugmeyer** [comment]
IMO it would be simpler to completely remove `enum` (`ErrorSource`, `StatusType`, `HandlerType`...) and use the string instead. Or, we could also use plain constants like:
```ts
const SOURCE_AGENT = 'agent' as const
```
This would result in a smaller code output, and avoid having two "types" for the same values (ex: `ErrorSource` and `Source`, `HandlerType` and `Handler`...)

 WDYT?

**Review by BenoitZugmeyer — APPROVED** [review]
LGTM!

**Comment by aleksandrlat** [comment]
Oh no why did you remove types? We need to duplicate type on our own now :(

**Comment by bcaudan** [comment]
Hi @aleksandrlat,
What was your usage?

**Pull request merged**